docs/specification/draft/basic/security_best_practices.mdx

```diff
@@ -0,0 +1,106 @@
+---
+title: Security Best Practices
+---
+
+## 1. Introduction
+
+### 1.1 Purpose and Scope
+
+This document provides security considerations for the Model Context Protocol (MCP), complementing the MCP Authorization specification. This document identifies security risks, attack vectors, and best practices specific to MCP implementations.
+
+The primary audience for this document includes developers implementing MCP authorization flows, MCP server operators, and security professionals evaluating MCP-based systems. This document should be read alongside the MCP Authorization specification and [OAuth 2.0 security best practices](https://datatracker.ietf.org/doc/html/rfc9700).
+
+## 2. Attacks and Mitigations
+
+This section gives a detailed description of attacks on MCP implementations, along with potential countermeasures.
+
+### 2.1 Confused Deputy Problem
+
+An attacker can exploit configurations where an MCP server operates as a proxy in front of another resource server, leading to the confused deputy problem.
+
+#### 2.1.1 Terminology
+
+**MCP Proxy Server**
+: An MCP server that acts as an intermediary between MCP clients and a protected 
+  third-party API. The MCP proxy server provides MCP functionality while delegating
+API operations to a third-party API server.  The MCP proxy server acts as a single OAuth client to the third-party API server.
+
+**Third-Party Authorization Server**
+: The authorization server that protects access to the third-party API. This server may not
+  support dynamic client registration, requiring the MCP proxy server to use a single static 
+  client ID for all requests.
+
+**Third-Party API**
+: The protected resource server that provides the actual API functionality. Access to this 
+  API requires tokens issued by the third-party authorization server.
+
+**Static Client ID**
+: A fixed OAuth 2.0 client identifier used by the MCP proxy server when communicating with
+  the third-party authorization server, shared across all MCP clients.
```

Comment:
I am not sure "shared across all MCP clients" is the right framing here. It's used by the server to request access tokens _for the server_ (I don't think we're talking about them being passed upstream to the client)?

Reply:
ah I see, yea I meant "shared" as in "used for all of them", not necessarily exposed to them, will edit